File: sample/scala-spring-boot-mvc/src/main/scala/com/benjaminsproule/sample/scala/SampleController.scala
package com.benjaminsproule.sample.scala

import io.swagger.annotations.{Api, ApiOperation}
import org.springframework.web.bind.annotation.{RequestMapping, RequestMethod, RestController}

@RestController
@Api(value = "/sample", description = "Sample REST for Integration Testing")
class SampleController {

    @RequestMapping(method = Array(RequestMethod.GET), path = Array("/sample"), produces = Array("application/json"))
    @ApiOperation(value = "Return hello message", response = classOf[String])
    def home: String = {
        "{\"Hello\": \"World!\"}"
    }
}

File: sample/kotlin-spring-boot-mvc/src/main/kotlin/com/benjaminsproule/sample/kotlin/SampleController.kt
package com.benjaminsproule.sample.kotlin

import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController

@RestController
@Api(value = "/sample", description = "Sample REST for Integration Testing")
class SampleController {

    @RequestMapping(method = [RequestMethod.GET], path = ["/sample"], produces = ["application/json"])
    @ApiOperation(value = "Return hello message", response = String::class)
    fun home(): String {
        return "{\"Hello\": \"World!\"}"
    }
}
